proposals/0448-type-variable-scoping.rst

```diff
@@ -198,42 +233,74 @@ All other variables are explicitly bound, and the inconsistency means more to le
 What are tedious tasks for the expert may be helpful learning steps to them.
 
-Further, the most beginnning students may be taught with both ``-XNoImplicitForAll`` and ``-XNoExplicitForAll``.
+Further, most beginning students may be taught with both ``-XImplicitBinds``, ``-XNoExplicitForAll``, and ``-XNoPolyKinds``.
 This means it's impossible to write forall types by any means.
 Combine with ``-Wmissing-signatures`` and ``-Wmissing-local-signatures``, so inferred polymorphic types of bindings are also prohibited, and a monomorphic custom prelude, and forall types are all but expunged entirely.
 
-I don't wish to argue whether these choices do or don't actually help learning, but just state that some people have opinions that they do and there is no technical reason GHC cannot accommodate them.
+@Ericson2314 doesn't wish to argue whether these choices do or don't actually help learning, but just state that some people have opinions that they do and there is no technical reason GHC cannot accommodate them.
+
+Consistency
+^^^^^^^^^^^
+
+Notice how today that out-of-scope variables in negative position signatures are implicitly bound in *different* ways depending on whether they are type variables (in pattern signatures) or kind variables (in negative position kind signatures).
+By banning implicit binding, we side-step that difference.
+
+After all, given::
+
+  data Foo (a :: k)
+
+desugars to::
+
+  data Foo @k (a :: k)
+
+a new Haskeller might conceivably think::
+
+  \(Foo (a :: k) -> ..
+
+desugars to::
+
+  \(Foo @k (a :: k) ->
+
+or::
+
+  \ @k (Foo (a :: k) ->
+
+which happen to be true in some simple common cases, but are in fact incorrect in general.
+
+That it takes a complicated example to show why these false desugarings aren't true in general make this is a huge educational stumbling block!
 
 Unified Namespace
 ^^^^^^^^^^^^^^^^^
 
 If `#270`_ is accepted, there will be a way to program Haskell with "morally" one namespace for types and terms alike.
-However, there is one exception to the unification of namespaces: lower case variables in type signatures bound "like terms" still are treated as free and implicitly bound with a ``forall`` instead::
+However, there is one exception to the unification of namespaces: lower case variables in type signatures bound "like terms" still are treated as free and implicitly bound instead::
 
   t = Int
-  x :: t -- sugar for 'forall t. t', not 't ~ Int'
+  x :: t -- sugar for 'forall t. t', not a use of 't' resolving to 'Int'
   x = 0
 
-Should the ``t`` in ``x :: t`` cause an implicit ``forall t.`` to be synthesized or not? With ``-XNoImplicitForAll``, we know
-it will not, and thus can refer to the ``t`` defined above, once such a reference is possible (left to another proposal).
+  t = Int
+  foo (x :: t) = 0 -- sugar for 'foo = let t = _ in \(x :: t) -> 0'
+
+Should the ``t`` in each ``x :: t`` cause implicit ``forall t.`` and ``let t = _ in`` to be synthesized or not?
```

Comment:
.

Reply:
The sentence that appears removed is on the line below.